miner/worker.go

```diff
@@ -1090,7 +1100,11 @@ func (w *worker) generateWork(params *generateParams) (*types.Block, error) {
 			log.Warn("Block building is interrupted", "allowance", common.PrettyDuration(w.newpayloadTimeout))
 		}
 	}
-	return w.engine.FinalizeAndAssemble(w.chain, work.header, work.state, work.txs, work.unclelist(), work.receipts)
+	block, err := w.engine.FinalizeAndAssemble(w.chain, work.header, work.state, work.txs, work.unclelist(), work.receipts)
+	if err != nil {
+		return nil, nil, err
+	}
+	return block, totalFees(block, work.receipts), nil
```

Comment:
Could you also pull in the change that makes total fees return a big.int? It's way better to work with. 

```go
func totalFees(block *types.Block, receipts []*types.Receipt) *big.Int {
	feesWei := new(big.Int)
	for i, tx := range block.Transactions() {
		minerFee, _ := tx.EffectiveGasTip(block.BaseFee())
		feesWei.Add(feesWei, new(big.Int).Mul(new(big.Int).SetUint64(receipts[i].GasUsed), minerFee))
	}
	return feesWei
```
This will save us a bit of headache down the line when we need to cast back to wei for the api in the future

Reply:
Yep, will do it

Reply:
`minerFee, _ := tx.EffectiveGasTip(block.BaseFee())` -- ignoring the error there means the `minerFee` can be negative (and the tx invalid). Hopefully, at this point, the tx is already validated against the basefee. Otherwise this can be attacked, so I just wanted to point it out.

Reply:
This is the code that we already use in the miner to calculate the fee for a block. It just changes the type of the result from big.float to big.int

Reply:
pushed